src/lib/Libattr/attr_func.c

```diff
@@ -113,34 +90,68 @@ clear_attr(attribute *pattr, struct attribute_def *pdef)
 
 /**
  * @brief
- * 	find_attr - find attribute definition by name
+ * 	Create the search index for the provided attribute def array
+ *
+ * @param[in] attr_def - ptr to attribute definitions
+ * @param[in] limit - limit on size of def array
+ *
+ * @return	void *
+ * @retval	 NULL	Failure
+ * @retval	!NULL	The address of the search index created
+ *
+ */
+void *
+cr_attrdef_idx(struct attribute_def *adef, int limit)
+{
+	int i;
+	void *attrdef_idx = NULL;
+
+	if (!adef)
+		return NULL;
+
+	/* create the attribute index */
+	if ((attrdef_idx = pbs_idx_create(PBS_IDX_CASE_CMP, 0)) == NULL)
+		return NULL;
+
+	/* add all attributes to the tree with key as the attr name */
+	for (i = 0; i < limit; i++) {
+		if (pbs_idx_insert(attrdef_idx, adef->at_name, adef) != PBS_IDX_RET_OK) {
+			/* destroy this index */
```

Comment:
we are not destroying the index.

Reply:
yeah i will remove this comment - we would quit PBS at this point, so no point destroying the index